src/components/EventUserTable.js

```diff
@@ -27,25 +40,111 @@ export class EventUserTable extends Component {
     this.getEventTableData(this.props.event.id);
   }
 
-  async getEventTableData(eventID) {
+  /* updates stats and the rows in the table
+     faculty, gender, dietary, and year stats are only computed on the initial render of the component
+     # of registered/checkedin etc. is computed every single time this function is called
+  */
+  async getEventTableData(eventID, firstRender) {
+
     const params = new URLSearchParams({
+      eventID: eventID
+    })
+
+    await fetchBackend('/registration/scanEvent?' + params, 'GET')
+      .then(response => response.json())
+      .then(response => {
+        let heardFrom = {};
+        for (let i = 0; i < response.size; ++i) {
+          const temp = response.data[i];
+          if (temp.hasOwnProperty('heardFrom')) {
+            heardFrom[temp.heardFrom] = heardFrom.hasOwnProperty(temp.heardFrom) ? heardFrom[temp.heardFrom] + 1 : 1;
+          }
+        }
+        this.setState({ heardFrom })
+      })
+
+    const eventParams = new URLSearchParams({
       id: eventID
     });
 
-    fetchBackend('/events/getUsers?' + params, 'GET')
+    fetchBackend('/events/getUsers?' + eventParams, 'GET')
       .then(response => response.json())
-      .then(response => {
-        this.setState({ rows: response });
+      .then(async users => {
+        let obj = {
+          'registered': 0,
+          'checkedIn': 0,
+          'waitlisted': 0,
+          'cancelled': 0,
+          'faculties': {},
+          'years': {},
+          'dietary': {},
+          'genders': {},
+          'heardFrom': {}
+        };
+
+        users.forEach(user => {
+          if (user.hasOwnProperty('registrationStatus')) {
+            switch (user.registrationStatus) {
+              case REGISTRATION_STATUS.REGISTERED:
+                ++obj['registered'];
+                break;
+              case REGISTRATION_STATUS.CHECKED_IN:
+                ++obj['checkedIn'];
+                break;
+              case REGISTRATION_STATUS.WAITLISTED:
+                ++obj['waitlisted'];
+                break;
+              case REGISTRATION_STATUS.CANCELLED:
+                ++obj['cancelled'];
+                break;
+              default:
+                return;
```

Comment:
And maybe use your one line solution that you used for faculty, diet etc instead of a switch

Reply:
i think it would be about the same number of lines since it would be an `if` statement + the one liner thing for each REGISTRATION_STATUS.CONSTANT

Reply:
but it also makes the code more consistent so idk